hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java

```diff
@@ -107,17 +107,29 @@ public static void setNMWebAppHostNameAndPort(Configuration conf,
    */
   public static <T, R> R execOnActiveRM(Configuration conf,
       ThrowingBiFunction<String, T, R> func, T arg) throws Exception {
-    int haIndex = 0;
+    int activeRMIndex = 0;
+    int rmCount = 1;
+
     if (HAUtil.isHAEnabled(conf)) {
```

Comment:
What if we try some early return pattern?

```
if (!HAUtil.isHAEnabled(conf)) {
  // If i understand right here we have just 1 RM so we can just try with that
  String rmAddress = getRMWebAppURLWithScheme(conf, 0);
  return func.apply(rmAddress, arg);
}
// try RMHAUtils.findActiveRMHAId(conf)
// if null do the iteration
// finally apply and return
```

Maybe will be a bit easyer to read cause now we have the 
```
int activeRMIndex = 0;
int rmCount = 1;
```
code what related to non HA, than HA code, than some mix.